server/src/main/java/org/candlepin/dto/DTOTranslator.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto;
+
+import org.candlepin.model.ModelEntity;
+
+
+
+/**
+ * The DTOTranslator interface defines a set of standard functionality to be provided by all
+ * translator instances
+ *
+ * @param <I>
+ *  The class of input model objects this translator supports
+ *
+ * @param <O>
+ *  The class of DTO objects output by this translator
+ */
+public interface DTOTranslator<I extends ModelEntity, O extends CandlepinDTO> {
+
+    /**
+     * Translates the given source object to a DTO, omitting any nested model objects.
+     *
+     * @param source
+     *  The source object to translate
+     *
+     * @throws IllegalArgumentException
+     *  if source is null
+     *
+     * @return
+     *  a new DTO representing the translated object, or null if the source object is null
+     */
+    CandlepinDTO translate(ModelEntity<I> source);
+
+    /**
+     * Translates the given source object to a DTO, using the provided DTO factory to translate any
+     * nested model objects. If a factory is not provided, then any nested model objects will be
+     * omitted from the resultant DTO.
+     *
+     * @param factory
+     *  The DTO factory to use for translating nested objects in the source object
+     *
+     * @param source
+     *  The source object to translate
+     *
+     * @throws IllegalArgumentException
+     *  if source is null
+     *
+     * @return
+     *  a new DTO representing the translated object, or null if the source object is null
+     */
+    CandlepinDTO translate(DTOFactory factory, ModelEntity<I> source);
+
+    /**
+     * Populates the given DTO object with data from the source object, omitting any nested model
+     * objects, setting their respective fields to null.
+     *
+     * @param source
+     *  The model object to translate
+     *
+     * @param destination
+     *  The DTO to populate with data from the source object
+     *
+     * @throws IllegalArgumentException
+     *  if either source or destination are null or not subclasses of the proper classes
+     *
+     * @return
+     *  The populated DTO
+     */
+    CandlepinDTO populate(ModelEntity<I> source, CandlepinDTO<O> destination);
```

Comment:
Not sure, but due to the implementation, it was virtually zero effort to add, so I did.